### PR TITLE
misc: use Vector2D, Box and Mat3x3 from hyprutils

### PR DESCRIPTION
**Pull request by PaideiaDilemma**
Switch to hyprutils for Vector2D, Box and Mat3x3.

### CODE REVIEW

File: src/core/LockSurface.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 static void handleConfigure(void* data, ext_session_lock_surface_v1* surf, uint32_t serial, uint32_t width, uint32_t height) {
     const auto PSURF = (CSessionLockSurface*)data;
-    PSURF->configure({width, height}, serial);
+    PSURF->configure({(double)width, (double)height}, serial);
```
Comment (PaideiaDilemma) [inline src/core/LockSurface.cpp:10]: This ok?

File: src/helpers/Math.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+#include <wayland-client.h>
+
+#include <hyprutils/math/Box.hpp>
+#include <hyprutils/math/Vector2D.hpp>
+#include <hyprutils/math/Mat3x3.hpp>
+
+using namespace Hyprutils::Math;
+
+eTransform          wlTransformToHyprutils(wl_output_transform t);
+wl_output_transform invertTransform(wl_output_transform tr);
```
Comment (PaideiaDilemma) [inline src/helpers/Math.hpp:12]: Same as in hyprland. I think we don't want to move `wlTransformToHyprutils` and `invertTransform` to hyprutils, as that would mean we have to get the `WL_OUTPUT_TRANSFORM*` definitions from somewhere right?

Also `Math.hpp` is included for Box, Vector2D and Mat3x3. But clangd complains that the header is not used directly. Let me know if we want the includes to work differently.

File: src/renderer/widgets/IWidget.hpp [hunk 3]
```diff
@@ -1,8 +1,10 @@
 #pragma once
 
-#include "../../helpers/Vector2D.hpp"
+#include "../../helpers/Math.hpp"
 #include <string>
 
+#define VEC2CONVERT(vec) Vector2D(vec.x, vec.y)
+
```
Comment (PaideiaDilemma) [inline src/renderer/widgets/IWidget.hpp:7]: This is just temporary.
What would be the best way to convert Hyprlang::VEC2 to Vector2D in all the widgets?

File: src/renderer/widgets/IWidget.cpp [hunk 4]
```diff
@@ -20,12 +20,18 @@ namespace std {
 }
 #endif
 
+Vector2D rotateVector(const Vector2D& vec, const double& ang) {
+    const double COS = std::abs(std::cos(ang));
+    const double SIN = std::abs(std::sin(ang));
+    return Vector2D(vec.x * COS + vec.y * SIN, vec.x * SIN + vec.y * COS);
+}
+
```
Comment (PaideiaDilemma) [inline src/renderer/widgets/IWidget.cpp:28]: We could move that to hyprutils I guess.

File: src/core/LockSurface.cpp (hunk 1 above)
Comment (vaxerski) [reply src/core/LockSurface.cpp:10]: zis ok

File: src/helpers/Math.hpp (hunk 2 above)
Comment (vaxerski) [reply src/helpers/Math.hpp:12]: looks fine

File: src/renderer/widgets/IWidget.hpp (hunk 3 above)
Comment (vaxerski) [reply src/renderer/widgets/IWidget.hpp:7]: write a custom ctor out-of-line? is that allowed in C++? if not, that or a fn.

File: src/renderer/widgets/IWidget.cpp (hunk 4 above)
Comment (vaxerski) [reply src/renderer/widgets/IWidget.cpp:28]: yes, but for now it has to stay here

File: src/renderer/widgets/IWidget.hpp (hunk 3 above)
Comment (PaideiaDilemma) [reply src/renderer/widgets/IWidget.hpp]: you can't sadly. I added a fn.

**Review by vaxerski — APPROVED** [review]
lgtm

**Comment by vaxerski** [comment]
anything else?

**Comment by PaideiaDilemma** [comment]
hmm nah i think this is it.

**Comment by PaideiaDilemma** [comment]
ahh wait maybe add `hyprutils>=0.2.3` to the readme

**Comment by vaxerski** [comment]
nah who cares about the readme

**Comment by PaideiaDilemma** [comment]
not me xD

**Comment by vaxerski** [comment]
I've doen it myslfe

**Pull request merged**